[jloh02] iP

jloh02 opened the pull request:
## Vuewee
A fully integrated application to store manage tasks
> So you'll never forget that assignment due 2359 tonight

### Features
- Add **not 1 but 3** different kinds of tasks!
- Mark tasks as done/not done
- Persistent storage - Keep your data even after closing the application

### Usage
#### Installing Vuewee
Download Vuewee [here](https://github.com/jloh02/ip/releases/latest)

#### Commands
Add a todo task, deadline task or event:
`todo <taskName>`
`deadline<taskName> /by <yyyy-MM-dd>`
`event <taskName> /from <yyyy-MM-dd> /to <yyyy-MM-dd>`

To show the task list
`list`

To mark a task as done/not done:
`mark <taskNo>`
`unmark <taskNo>`

To exit:
`bye`

### Features to add
- [ ] Create UI using Java FX
- [ ] Improve UX

### Additional Remarks
This is the current main method
```java
package vuewee;

import java.util.Scanner;

public class Vuewee {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        TaskListUI ui = new TaskListUI(scanner);
        ui.run();
        scanner.close();
    }
}

```

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

jloh02 committed:
rename bot, greet and exit

jloh02 committed:
add echo function

jloh02 committed:
add functionality to add to list and display list

jloh02 committed:
add Task.java

jloh02 committed:
Add .vscode to .gitignore

jloh02 committed:
use Task class for handling task status

jloh02 committed:
use success status for marking and unmarking

jloh02 committed:
Refactor into switch case

jloh02 committed:
Handle invalid task numbers

jloh02 committed:
Use TodoTask

jloh02 committed:
Add event and deadline tasks

jloh02 committed:
Print message to error stream

jloh02 committed:
Use TaskList class to abstract add task messages

jloh02 committed:
Only print to stdout

jloh02 committed:
Add text test case

jloh02 committed:
Update tests

jloh02 committed:
Use exceptions for errors

jloh02 committed:
Use TaskListException superclass

jloh02 committed:
Add delete function

jloh02 committed:
Rename NoTaskParametersFoundException for clarity

jloh02 committed:
Refactor to use ArrayList instead of static array

jloh02 committed:
Handle missing event in description

jloh02 committed:
Fix index issues

jloh02 committed:
Update tests

jloh02 committed:
Use enum for TaskType

jloh02 committed:
Create command parser

jloh02 committed:
Handle integer parsing in parser

jloh02 committed:
Refactor exceptions

jloh02 committed:
Use enum for command type

jloh02 committed:
Refactor into TaskListUI

jloh02 committed:
Add data folder into gitignore

jloh02 committed:
Create task store to store data as txt

jloh02 committed:
Fix tests for storage data

jloh02 committed:
Add generic parsers

jloh02 committed:
Refactor to pass functions and dates

jloh02 committed:
Use command abstract class

jloh02 committed:
Update naming

jloh02 committed:
Organize into packages

jloh02 committed:
Update tests

jloh02 committed:
Fix DeleteCommand

jloh02 committed:
Merge branch 'add-gradle-support' into branch-A-Gradle

jloh02 committed:
Update gradle build class name

jloh02 committed:
Fix Task serialization and deserialization

jloh02 committed:
Add unit tests for command package
* Since command package testing involves all components including the parsers, all cases can be captured
* Unit tests do not cover the TaskListUI which have to use integration testing

jloh02 committed:
Add JavaDocs for command package

jloh02 committed:
Update jar archive base name

jloh02 committed:
Add remaining javadocs

jloh02 committed:
Merge branch 'branch-A-JavaDoc'

jloh02 committed:
Update boolean names

jloh02 committed:
Fix coding style

jloh02 committed:
Merge branch 'branch-A-CodingStandard'

jloh02 committed:
Add find feature

jloh02 committed:
Fix tests

jloh02 committed:
Fx import orders

choonzies commented on src/main/java/vuewee/TaskListUI.java:
```diff
@@ -0,0 +1,129 @@
+    }
+
+    // Simple helper method to determine if the task count is 1 or more
+    private String taskWord() {
```
Perhaps the name of the method `taskWord` may be ambiguous? 🤔

choonzies commented on src/main/java/vuewee/TaskListUI.java:
```diff
@@ -0,0 +1,129 @@
+                System.out.println("____________________________________________________________");
+                storage.storeTasks(this.taskList);
+            }
+        } catch (EndProgramException e) {
```
Empty catch block present here, perhaps there can be a way to handle these exceptions such as print the issue out for the user?

choonzies commented on src/main/java/vuewee/command/DeadlineCommand.java:
```diff
@@ -0,0 +1,30 @@
+class DeadlineCommand extends Command {
+    /**
+     * Executes the deadline command by parsing the input, creating a new deadline
+     * task, and adding it to the task list
```
I think this first sentence of the JavaDoc can end with a fullstop 😄

choonzies commented on src/main/java/vuewee/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,23 @@
+     *
+     * @param ui       The user interface for displaying the task list.
+     * @param taskList The task list containing the tasks.
+     * @param parser   The command parser for parsing user input.
```
Perhaps the first sentence should just be a summary sentence instead of 2 sentences? The potential exception thrown can put under the "\@throws".

choonzies commented on src/main/java/vuewee/parser/CommandOption.java:
```diff
@@ -0,0 +1,94 @@
+    /**
+     * Parses the input string for the option value and saves the parsed value.
+     *
+     * @param input        the input string to parse
```
Nit, spacing issue.

choonzies commented on src/main/java/vuewee/parser/CommandParser.java:
```diff
@@ -0,0 +1,123 @@
+
+        // Parse options
+        // All options should start with /option_name <option_value>
+        int minStartMatch = this.argument.length();
```
I think the naming of variable can be better here?

choonzies commented on src/main/java/vuewee/task/DeadlineTask.java:
```diff
@@ -0,0 +1,69 @@
+ * Represents a task with a deadline.
+ */
+public class DeadlineTask extends Task {
+    private TaskLocalDate by;
```
I think variable naming can be better here, perhaps deadlineDate/byDate instead?

choonzies commented on src/main/java/vuewee/task/DeadlineTask.java:
```diff
@@ -0,0 +1,69 @@
+            throw new IllegalArgumentException("Invalid task format: " + serializedTask);
+        }
+
+        boolean isDone = parts[0].equals("1");
```
Good intuitive naming of boolean variable!

choonzies commented on src/main/java/vuewee/task/EventTask.java:
```diff
@@ -0,0 +1,68 @@
+ */
+public class EventTask extends Task {
+    private TaskLocalDate from;
+    private TaskLocalDate to;
```
Similarly, perhaps variable naming can be better here?

choonzies commented on src/main/java/vuewee/task/TaskList.java:
```diff
@@ -0,0 +1,131 @@
+     * @return true if the task was successfully marked, false otherwise
+     * @throws IndexOutOfBoundsException if the index is out of range
+     */
+    public boolean markTask(int taskNumber, boolean done) throws IndexOutOfBoundsException {
```
`done` variable can be improved? 😄

choonzies reviewed (commented):
LGTM. Mainly minor nits on variable naming.

jloh02 committed:
Add checkstyle config and fix styles

jloh02 committed:
Merge branch 'branch-A-CheckStyle'

TheRareFox commented on src/main/java/vuewee/TaskListUI.java:
```diff
@@ -0,0 +1,173 @@
+     */
+    public void displayTasks(TaskList tasks) throws IllegalCommandException {
+        if (tasks.size() == 0) {
+            throw new IllegalCommandException("No tasks found.");
```
Would it be better to print no tasks found rather than throwing an error?

TheRareFox reviewed (commented):
LGTM. Not much things to nit about.

jloh02 committed:
Create abstract TaskListUi class

jloh02 committed:
Add basic GUI

jloh02 committed:
Add interactive GUI for bot to send messages

jloh02 committed:
Add handler for user input and bot commands

jloh02 committed:
Change JavaFX to use FXML

jloh02 committed:
Update by date help

jloh02 committed:
Fix unit tests

jloh02 committed:
Fix code style

jloh02 committed:
Merge branch 'branch-Level-10'

jloh02 committed:
Fix EventTask deserialization

jloh02 committed:
Improve GUI aesthetics
- Use CSS to style JavaFX
- Style as per JavaFX tutorial step 5
- Update images

jloh02 committed:
Add assertions

jloh02 committed:
Merge pull request #1 from jloh02/branch-A-Assertions

Add assertions

jloh02 committed:
Rename methods to fix code quality
- Change Command::execute to Command::executeCommand for clarity

jloh02 committed:
Refactor for parsers to return descriptions of generic type
- Currently only support int and string descriptions
- Created DescriptionParser class to allow generic types for descriptions

jloh02 committed:
Fix code style and comments

jloh02 committed:
Merge branch 'master' of https://github.com/jloh02/ip into branch-A-CodeQuality

jloh02 committed:
Merge pull request #2 from jloh02/branch-A-CodeQuality

Improve code quality

jloh02 committed:
Fix assertion tests

jloh02 committed:
Add gradle.yml

jloh02 committed:
Remove IO redirected text tests

jloh02 committed:
Only run on push to master and all PRs

jloh02 committed:
Merge pull request #3 from jloh02/branch-A-CI

Add CI

jloh02 committed:
Add schedule command

jloh02 committed:
Fix tests

jloh02 committed:
Add list of commands for help

jloh02 committed:
Merge pull request #4 from jloh02/B-ViewSchedules

Add schedules command

jloh02 committed:
Set name to Vuewee

jloh02 committed:
Fix header comments

jloh02 committed:
Update docs

jloh02 committed:
Update Ui.png
